File: easy/3423. Maximum Difference Between Adjacent Elements in a Circular Array/3423. Maximum Difference Between Adjacent Elements in a Circular Array.ts
function maxAdjacentDistance(nums: number[]): number {
    const n = nums.length
    let maxDiff = Math.abs(nums[0] - nums[n - 1])

    for (let i = 0; i < n - 1; i++) {
        maxDiff = Math.max(maxDiff, Math.abs(nums[i] - nums[i + 1]))
    }

    return maxDiff
};

File: easy/3423. Maximum Difference Between Adjacent Elements in a Circular Array/3423. Maximum Difference Between Adjacent Elements in a Circular Array.py
from typing import List


class Solution:
    def maxAdjacentDistance(self, nums: List[int]) -> int:
        n = len(nums)
        maxDiff = 0

        for i in range(2 * n):
            maxDiff = max(maxDiff, nums[i % n] - nums[(i + 1) % n])

        return maxDiff
